docs/devenv_rhel8_auto.md

```diff
@@ -0,0 +1,56 @@
+# Automating Development Environment Setup
+This document describes how to automate the setup of MicroShift development environment per instructions in the following sections:
+* [Create Development Virtual Machine](./devenv_rhel8.md#create-development-virtual-machine)
+* [Build MicroShift](./devenv_rhel8.md#build-microshift)
+* [Run MicroShift Executable](./devenv_rhel8.md#run-microshift-executable)
+
+It is recommended to review the above mentioned documentation sections before proceeding with the automation instructions in the current document.
+
+## Create Virtual Machine
+Log into the hypervisor host and run the `create-vm.sh` script without arguments to see its usage.
+```bash
+$ ./scripts/devenv-builder/create-vm.sh 
+Usage: create-vm.sh <vm_name> <vm_disk_dir> <rhel_dvd_iso_file> <ncpus> <memory_in_GB> <disk_in_GB> <data_vol_size_in_GB>
+
+ERROR: Invalid number of arguments
+```
+> It is mandatory to use the RHEL 8 DVD (not boot) image for installing the virtual machine.
+
+As an example, run the following command to create a virtual machine named `microshift-dev` with 4 CPUs, 6GB of RAM, 50GB of total disk space and 2GB disk allocated to the data volume.
+```bash
+./scripts/devenv-builder/create-vm.sh microshift-dev ~/VirtualMachines/ ~/VirtualMachines/rhel-8.6-x86_64-dvd.iso 4 6 50 2 
```

Comment:
I know this is just an example, but the folder where the rhel image and the microshift-dev disk are going to be located must be configured to work properly with libvirt. I had to use the default storage pool (`/var/lib/libvirt/images/`) or otherwise I'd get these errors:

```
Starting install...
Retrieving file vmlinuz...                                                                                                     |  10 MB  00:00:00
Retrieving file initrd.img...                                                                                                  |  80 MB  00:00:00
Allocating 'microshift-dev.qcow2'                                                                                              |  50 GB  00:00:07
Removing disk 'microshift-dev.qcow2'                                                                                           |    0 B  00:00:00
ERROR    Cannot access storage file '/home/oglok/VirtualMachines/microshift-dev.qcow2' (as uid:107, gid:107): Permission denied
Domain installation does not appear to have been successful.
If it was, you can restart your domain by running:
  virsh --connect qemu:///system start microshift-dev
otherwise, please restart your installation.
```